certbot/renewal.py

```diff
@@ -301,7 +301,10 @@ def _renew_describe_results(config, renew_successes, renew_failures,
 def renew_all_lineages(config):
     """Examine each lineage; renew if due and report results"""
 
-    if config.domains != []:
+    # This is trivially False if config.domains is empty
+    if any(domain not in config.webroot_map for domain in config.domains):
```

Comment:
This is probably correct, though we should convince ourselves that there are no other ways that `config.domains` might get set, aside from the webroot plugin.  For instance, if there's a cli.ini file with `domains` set to we want to be erroring out here?